src/hooks/useRoom.ts

```diff
@@ -22,12 +21,11 @@ export const useRoom = ({ room, venueName }: UseRoomProps) => {
 
   const roomUrl = room?.url ?? "";
 
-  // @debt This selector relies on all venues in firebase being loaded into memory.. not very efficient
-  const venues = useSelector(orderedVenuesSelector);
+  const { relatedVenues } = useRelatedVenues({});
 
   const roomVenue = useMemo(
-    () => venues?.find((venue) => roomUrl.endsWith(`/${venue.id}`)),
-    [roomUrl, venues]
+    () => relatedVenues?.find((venue) => roomUrl.endsWith(`/${venue.id}`)),
```

Comment:
`relatedVenues` will always be a defined array, so no need for `?`:

```suggestion
    () => relatedVenues.find((venue) => roomUrl.endsWith(`/${venue.id}`)),
```